server/lib/src/idm/application.rs

```diff
@@ -0,0 +1,631 @@
+            (Attribute::Class, EntryClass::Object.to_value()),
+            (Attribute::Class, EntryClass::Account.to_value()),
+            (Attribute::Class, EntryClass::ServiceAccount.to_value()),
+            (Attribute::Class, EntryClass::Application.to_value()),
+            (Attribute::Class, EntryClass::Person.to_value()),
+            (Attribute::Name, Value::new_iname("test_app_name")),
+            (Attribute::Uuid, Value::Uuid(test_entry_uuid)),
+            (Attribute::Description, Value::new_utf8s("test_app_desc")),
+            (
+                Attribute::DisplayName,
+                Value::new_utf8s("test_app_dispname")
+            )
+        );
+
+        let ce = CreateEvent::new_internal(vec![e1]);
+        let cr = idms_prox_write.qs_write.create(&ce);
+        assert!(!cr.is_ok());
+
+        let test_entry_uuid = Uuid::new_v4();
+        let e1 = entry_init!(
+            (Attribute::Class, EntryClass::Object.to_value()),
+            (Attribute::Class, EntryClass::Account.to_value()),
+            (Attribute::Class, EntryClass::Application.to_value()),
+            (Attribute::Class, EntryClass::Person.to_value()),
+            (Attribute::Name, Value::new_iname("test_app_name")),
+            (Attribute::Uuid, Value::Uuid(test_entry_uuid)),
+            (Attribute::Description, Value::new_utf8s("test_app_desc")),
+            (
+                Attribute::DisplayName,
+                Value::new_utf8s("test_app_dispname")
+            )
+        );
+
+        let ce = CreateEvent::new_internal(vec![e1]);
+        let cr = idms_prox_write.qs_write.create(&ce);
+        assert!(!cr.is_ok());
+
+        let test_entry_uuid = Uuid::new_v4();
+        let e1 = entry_init!(
+            (Attribute::Class, EntryClass::Object.to_value()),
+            (Attribute::Class, EntryClass::Application.to_value()),
+            (Attribute::Name, Value::new_iname("test_app_name")),
+            (Attribute::Uuid, Value::Uuid(test_entry_uuid)),
+            (Attribute::Description, Value::new_utf8s("test_app_desc")),
+            (
+                Attribute::DisplayName,
+                Value::new_utf8s("test_app_dispname")
+            )
+        );
+
+        let ce = CreateEvent::new_internal(vec![e1]);
+        let cr = idms_prox_write.qs_write.create(&ce);
+        assert!(!cr.is_ok());
+
+        let test_entry_uuid = Uuid::new_v4();
+        let e1 = entry_init!(
+            (Attribute::Class, EntryClass::Object.to_value()),
+            (Attribute::Class, EntryClass::Application.to_value()),
+            (Attribute::Class, EntryClass::ServiceAccount.to_value()),
+            (Attribute::Name, Value::new_iname("test_app_name")),
+            (Attribute::Uuid, Value::Uuid(test_entry_uuid))
+        );
+
+        let ce = CreateEvent::new_internal(vec![e1]);
+        let cr = idms_prox_write.qs_write.create(&ce);
+        assert!(cr.is_ok());
+    }
+
+    #[idm_test]
+    async fn test_idm_application_no_linked_group(
```

Comment:
I thought linked group was required?

Reply:
Nope it wasn't commented on schema that it needs to be a must attribute.

Reply:
Discussion continues below.